pallets/runtime/tests/src/confidential_asset_test.rs

```diff
@@ -201,60 +235,149 @@ fn issuers_can_create_and_rename_confidential_tokens() {
-        let alice_id = register_keyring_account(AccountKeyring::Alice).unwrap();
+        let (scrt_account, mercat_account_tx) = gen_account(
+            0,          // Transaction id. Not important in this test.
+            [10u8; 32], // seed
+            &token_names[1][..],
+            valid_asset_ids,
+        );
 
-        // These are the encryptions keys used by MERCAT and are different from the signing keys
-        // that Polymesh uses.
-        let mut rng = StdRng::from_seed([10u8; 32]);
-        let elg_secret = ElgamalSecretKey::new(Scalar::random(&mut rng));
-        let elg_pub = elg_secret.get_public_key();
-        let enc_keys = EncryptionKeys {
-            pblc: elg_pub.into(),
-            scrt: elg_secret.into(),
-        };
+        ConfidentialAsset::validate_mercat_account(
+            Origin::signed(owner),
+            mercat_account_tx.clone(),
+        )
+        .unwrap();
 
-        // Positive test: use one of the registered asset ids.
-        let asset_id = AssetId {
-            id: *Ticker::try_from(&token_names[1][..]).unwrap().as_bytes(),
-        };
-        let asset_id_witness = CommitmentWitness::from((asset_id.clone().into(), &mut rng));
-        let scrt_account = SecAccount {
-            enc_keys,
-            asset_id_witness,
+        // ------------- START: Computations that will happen in Alice's Wallet ----------
+        let amount: u32 = token.total_supply; // mercat amounts are 32 bit integers.
+        let mut rng = StdRng::from_seed([42u8; 32]);
+        let issuer_account = Account {
+            scrt: scrt_account,
+            pblc: mercat_account_tx.pub_account.clone(),
         };
 
-        let valid_asset_ids = convert_asset_ids(valid_asset_ids);
-        let account_id = 0;
-        let mercat_tx_id = 0;
-        let mercat_account_tx = AccountCreator {}
-            .create(
-                mercat_tx_id,
-                &scrt_account,
-                &valid_asset_ids,
-                account_id,
+        let initialized_asset_tx = AssetIssuer {}
+            .initialize_asset_transaction(
+                1, // Transaction id. Not important in this test.
+                &issuer_account,
+                &[],
+                amount,
                 &mut rng,
             )
             .unwrap();
 
         // ------------- END: Computations that will happen in the Wallet ----------
 
+        // Wallet submits the transaction to the chain for verification.
+        ConfidentialAsset::mint_confidential_asset(
+            Origin::signed(owner),
+            ticker,
+            amount.into(), // convert to u128
+            initialized_asset_tx,
+        )
+        .unwrap();
+
+        // ------------------------- Ensuring that the asset details are set correctly
+        // Check the update investor count for the newly created asset.
+        //assert_eq!(Statistics::investor_count_per_asset(ticker), 1); // TODO not sure why this fails!
```

Comment:
ditto

Reply:
Once you mint the asset you have to make sure to call the `update_transfer_stats` function of the statistic pallet.